src/sentry/search/events/fields.py

```diff
@@ -1314,6 +1314,53 @@ def with_default(default, argument):
     return argument
 
 
+# Implementation of certain FunctionArg subclasses
+# to validate aggregate functions implemented in SnQL.
+class SnQLColumn(FunctionArg):
+    def __init__(self, name, allowed_columns=None):
+        super().__init__(name)
+        self.allowed_columns = allowed_columns or []
+
+    def normalize(self, value, _):
+        if len(self.allowed_columns) > 0:
+            if value in self.allowed_columns:
+                return value
+            else:
+                raise InvalidFunctionArgument(f"{value} is not an allowed column")
+
+        snuba_column = SEARCH_MAP.get(value)
+        if not snuba_column:
+            raise InvalidFunctionArgument(f"{value} is not a valid column")
+
+        return value
+
+
+class SnQLStringArg(FunctionArg):
```

Comment:
Does this differ from the existing `StringArg` implementation? Why do we need a custom `SnQLStringArg`?

Reply:
It's the return value - in the original we add quotes to denote it's a string arg, but with SnQL, we shouldn't

Reply:
Could we comment that with a TODO: explaining to replace StringArg with this class once we're converted?

Also thoughts on subclassing StringArg, so that if there's any bugs that need to be fixed during the migration its handled for both snql&not?

Reply:
Yeah, sure thing